server/src/main/java/org/elasticsearch/action/admin/cluster/snapshots/status/SnapshotStatus.java

```diff
@@ -71,14 +72,14 @@
     @Nullable
     private Boolean includeGlobalState;
 
-    SnapshotStatus(final Snapshot snapshot, final State state, final List<SnapshotIndexShardStatus> shards,
-                   final Boolean includeGlobalState) {
+    SnapshotStatus(Snapshot snapshot, State state, List<SnapshotIndexShardStatus> shards, Boolean includeGlobalState,
+                   long startTime, long time) {
         this.snapshot = Objects.requireNonNull(snapshot);
         this.state = Objects.requireNonNull(state);
         this.shards = Objects.requireNonNull(shards);
         this.includeGlobalState = includeGlobalState;
         shardsStats = new SnapshotShardsStats(shards);
-        updateShardStats();
+        updateShardStats(startTime, time);
```

Comment:
Can we check that they are not < 0?

Reply:
Actually, I don't think we need to I think :) We get these numbers from `org.elasticsearch.index.snapshots.IndexShardSnapshotStatus` which does all the checks internally so we should be good here?

Reply:
:+1: